api/v1alpha1/gateway_types.go

```diff
@@ -335,13 +346,23 @@ type ListenerExtensionObjectReference = ConfigMapsDefaultLocalObjectReference
 
 // GatewayStatus defines the observed state of Gateway.
 type GatewayStatus struct {
+	// Addresses lists the IP addresses that have actually been
+	// bound to the Gateway. These addresses may differ from the
+	// addresses in the Spec, e.g. if the Gateway automatically
+	// assigns an address from a reserved pool.
+	//
+	// These addresses should all be of type "IPAddress".
+	//
+	// +required
+	Addresses []GatewayAddress `json:"addresses" protobuf:"bytes,1,opt,name=addresses"`
```

Comment:
Although `spec` and `status` are two different sections, we should avoid naming a field exactly same in both for a reasonable user experience.
What if we rename the field in the `spec` as `RequestedAddress`?

Reply:
I'd rather discuss that in a separate issue for better visibility.

Reply:
Technically we should have Request... on all of the fields if this is the case, seems redundant.
Yes - let's discussion on a different issue.